crates/re_space_view_spatial/src/space_view_2d.rs

```diff
@@ -286,6 +297,96 @@ impl SpaceViewClass for SpatialSpaceView2D {
     }
 }
 
+fn count_non_nested_entities_with_component(
+    entity_bucket: &IntSet<EntityPath>,
+    subtree: &EntityTree,
+    component_name: &ComponentName,
+) -> usize {
+    if entity_bucket.contains(&subtree.path) {
+        // bool true -> 1
+        subtree.entity.components.contains_key(component_name) as usize
+    } else if !entity_bucket
+        .iter()
+        .any(|e| e.is_descendant_of(&subtree.path))
+    {
+        0
+    } else {
+        subtree
+            .children
+            .values()
+            .map(|child| {
+                count_non_nested_entities_with_component(entity_bucket, child, component_name)
+            })
+            .sum()
+    }
+}
+
+fn add_recommended_space_views_for_bucket(
+    ctx: &ViewerContext<'_>,
+    entity_bucket: &IntSet<EntityPath>,
+    recommended: &mut Vec<RecommendedSpaceView>,
+) {
+    // TODO(jleibs): Converting entity_bucket to a Trie would probably make some of this easier.
+    let tree = ctx.entity_db.tree();
+
+    // Find the common ancestor of the bucket
+    let root = EntityPath::common_ancestor_of(entity_bucket.iter());
+
+    // If the root of this bucket contains an image itself, this means the rest of the content
+    // is nested under some kind of 2d-visualizable thing. We expect the user meant to create
+    // a layered 2d space.
+    if entity_bucket.contains(&root) {
```

Comment:
The talk of images and 2d here comes out of left field s. There is nothing above setting the context that this is a function meant only for 2D stuff.